contracts/UFragments.sol

```diff
@@ -174,6 +189,42 @@ contract UFragments is ERC20Detailed, Ownable {
         return TOTAL_GONS;
     }
 
+    /**
+     * @return The number of successful permits by the specified address.
+     */
+    function nonces(address who) public view returns (uint256) {
+        return _nonces[who];
+    }
+
+    /**
+     * @return The underlying blockchain's chainID.
+     */
+    function getChainId() public view returns (uint256) {
```

Comment:
Does this need to be public?

Also, basic question but--does this being its own function use more gas than it would if it were just in the body of `DOMAIN_SEPARATOR()`?

Reply:
Hmm. Not necessarily. But if it is public, off-chain services can look up all the attributes to compute the domain_seperator from the token contract's interface.

```
keccak256(
    abi.encode(
        uFragments.EIP712_DOMAIN(),
        keccak256(bytes(uFragments.name())),
        keccak256(bytes(uFragments.EIP712_REVISION())),
        uFragments.getChainId(),
        uFragments.address
    )
)
```

Reply:
Makes sense.